app/src/main/java/org/simple/clinic/facility/change/FacilityListItem.kt

```diff
@@ -0,0 +1,49 @@
+package org.simple.clinic.facility.change
+
+import android.support.v7.util.DiffUtil
+import org.simple.clinic.facility.Facility
+
+data class FacilityListItem(
+    val facility: Facility,
+    val name: Name,
+    val address: Address
+) {
+
+  sealed class Name {
+    data class Highlighted(val name: String, val highlightStart: Int, val highlightEnd: Int) : Name()
+    data class Plain(val name: String) : Name()
+  }
+
+  sealed class Address {
+    data class WithStreet(val street: String, val district: String, val state: String) : Address()
+    data class WithoutStreet(val district: String, val state: String) : Address()
+  }
+
+  class Differ : DiffUtil.ItemCallback<FacilityListItem>() {
+    override fun areItemsTheSame(oldItem: FacilityListItem, newItem: FacilityListItem) = oldItem.facility.uuid == newItem.facility.uuid
+    override fun areContentsTheSame(oldItem: FacilityListItem, newItem: FacilityListItem) = oldItem == newItem
+  }
+
+  object Builder {
```

Comment:
What does everyone think of having a separate builder *object* for generating ui models? It's used as a static function instead of getting injected.